pkg/compose/compose.go

```diff
@@ -67,9 +67,19 @@ func getContainerNameWithoutProject(c moby.Container) string {
 func (s *composeService) Convert(ctx context.Context, project *types.Project, options api.ConvertOptions) ([]byte, error) {
 	switch options.Format {
 	case "json":
-		return json.MarshalIndent(project, "", "  ")
+		marshal, err := json.MarshalIndent(project, "", "  ")
+		if err != nil {
+			return nil, err
+		}
+		out := strings.ReplaceAll(string(marshal), "$", "$$")
+		return []byte(out), err
 	case "yaml":
-		return yaml.Marshal(project)
+		marshal, err := yaml.Marshal(project)
+		if err != nil {
+			return nil, err
+		}
+		out := strings.ReplaceAll(string(marshal), "$", "$$")
+		return []byte(out), err
```

Comment:
Probably wouldn’t optimize much, but this could just be
```suggestion
		out := bytes.ReplaceAll(marshal, []byte("$"), []byte("$$"))
		return out, nil
```
(and we could even extract to a function as it’s duplicated)